docs/content/guides/dagster-pipes/kubernetes.mdx

```diff
@@ -0,0 +1,248 @@
+
+At this point you can execute the rest of your Kubernetes code as normal, invoking various <PyObject object="PipesContext" module="dagster_pipes" /> APIs as needed.
+
+### Step 1.2: Define and build the container image
+
+Next, you'll package the script into a container image using a `Dockerfile`. For example:
+
+```dockerfile
+FROM python:3.10-slim
+
+pip install dagster-pipes
+
+COPY my_python_script.py .
+```
+
+Then, build the image:
+
+```shell
+docker build -t pipes-example:latest
+```
+
+**Note**: Depending on the Kubernetes setup you're using, you may need to upload the container image to a registry or otherwise make it available to the cluster. For example: `kind load docker-image pipes-example:latest`
+
+---
+
+## Step 2: Create the Dagster objects
+
+In this step, you’ll create a Dagster asset that, when materialized, opens a Dagster pipes session and spins up a Kubernetes pod to execute the container created in the previous step.
+
+### Step 2.1: Define the Dagster asset
+
+In your Dagster project, create a file named `dagster_k8s_pipes.py` and paste in the following code:
+
+```python
+# dagster_k8s_pipes.py
+
+from dagster import AssetExecutionContext, Definitions, asset
+from dagster_k8s import PipesK8sClient
+
+
+@asset
+def k8s_pipes_asset(context: AssetExecutionContext, k8s_pipes_client: PipesK8sClient):
+  return k8s_pipes_client.run(
+      context=context,
+      image="pipes-example:v1",
+  ).get_materialize_result()
+```
+
+Here’s what we did in this example:
+
+- Created an asset named `k8s_pipes_asset`
+
+- Provided <PyObject object="AssetExecutionContext" /> as the `context` argument to the asset. This object provides access to system APIs such as resources, config, and logging.
+
+- Specified a resource for the asset to use, <PyObject module="dagster_k8s" object="PipesK8sClient" />, which is a pre-built Dagster resource that allows you to quickly get Pipes working with your Kubernetes workspace.
+
+  We also specified the following for the resource:
+
+  - `context` - The asset's `context` (<PyObject object="AssetExecutionContext" />) data
+  - `image` - The Kubernetes image we created in [Step 1](#step-1-define-the-external-code-container)
+
+  These arguments are passed to the `run` method of <PyObject module="dagster_k8s" object="PipesK8sClient" />, which submits the provided cluster information to the Kubernetes API and then runs the specified `image`.
+
+- Returned a <PyObject object="MaterializeResult" /> object representing the result of execution. This is obtained by calling `get_materialize_result` on the <PyObject module="dagster_pipes" object="PipesClientCompletedInvocation" /> object returned by `run` after the execution in Kubernetes has completed. **Note**: Execution can take several minutes even for trivial scripts due to cluster provisioning times.
+
+<Note>
```

Comment:
I moved this up from the end of Step 1.2 since it felt buried, but will move it back if this should be done on the resource in `Definitions` and not here, for some reason.